wai-handler-hal.cabal

```diff
@@ -41,6 +41,7 @@ common opts
 
 common deps
   build-depends:
+    , aeson                 >=1.5.6.0 && <1.6 || >=2.0 && <2.2
```

Comment:
What's changed to make the main library require `aeson`? The PR summary only shows file moves.

Reply:
Right, this was added to try to pin a version of aeson in ci. I'll remove

Reply:
That caused CI to fail - i've reverted and if everything passes i'll add a comment referencing the upstream fix PR

Reply:
I would rather have failing CI (which should be temporary once `hal` gets fixed) than push an unnecessary dep (even a common one) onto downstream users.

We should probably think about reporting this to haskell-ci: it looks like it's considering only the library when installing dependencies. But why is it even _selecting_ `aeson-2.2` with current `hal`?

Reply:
Ah, right. Because `hal` declares no bounds on its dependencies. -_-

Reply:
Right, so it's `hal`'s fault, and the correct fix is a metadata revision, not to fake a pin by declaring a dependency we don't really need. Once you fix up the bounds in https://github.com/Nike-Inc/hal/pull/119 , please ask for metadata revisions. I can provide them if he's unresponsive (give it a week, I guess?). We'll have to revise any version containing that unparse function: [0.4.9](https://hackage.haskell.org/package/hal-0.4.9), [0.4.10](https://hackage.haskell.org/package/hal-0.4.10), [0.4.10.1](https://hackage.haskell.org/package/hal-0.4.10.1), [1.0.0](https://hackage.haskell.org/package/hal-1.0.0), [1.0.0.1](https://hackage.haskell.org/package/hal-1.0.0.1).

Reply:
Yeah, the dep is already sort of pushed onto users implicitly until that fix is made. I'm surprised their CI didn't fail for the same reasons

Reply:
I have two theories:

1. Because they're using Stackage (pinned global versions of packages) they haven't picked up new `aeson` yet and haven't noticed.
2. Because the dependency bounds are only on the `test-suite`, their development environment solves for the correct version of `aeson` but library consumers sometimes don't build tests.